yarn/assets/configuration/spec.yaml

```diff
@@ -4,6 +4,32 @@ files:
     options:
       - template: init_config
         options:
+          - name: collect_apps_all_states
+            description: |
+              Enable this parameter when using `collect_app_metrics` to collect metrics from applications of all states.
+              For example, collect metrics from applications with 'NEW', 'SUBMITTED', 'FINISHED', and other states.
+            value:
+              example: false
+              type: boolean
+          - name: collect_apps_states_list
+            description: |
+              Configure the application states of which to collect application metrics.
+            value:
+              type: array
+              example: &yarn_app_states_init
+              - 'ALL'
+              - 'NEW'
+              - 'NEW_SAVING'
+              - 'SUBMITTED'
+              - 'ACCEPTED'
+              - 'RUNNING'
+              - 'FINISHED'
+              - 'FAILED'
+              - 'KILLED'
+              default: *yarn_app_states_init
```

Comment:
you can omit the defaults I think as long as it's acceptable for the example to be the default